topdown/topdown.go

```diff
@@ -1932,7 +1907,34 @@ func evalTermsComprehension(t *Topdown, comp ast.Value, iter Iterator) error {
 		}
 
 		return Continue(t, comp, result, iter)
+	case *ast.SetComprehension:
+		result := ast.Set{}
+		child := t.Closure(comp.Body)
+
+		err := Eval(child, func(child *Topdown) error {
+			result = append(result, PlugTerm(comp.Term, child.Binding))
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
 
+		return Continue(t, comp, &result, iter)
+	case *ast.ObjectComprehension:
+		result := ast.Object{}
+		child := t.Closure(comp.Body)
+		err := Eval(child, func(child *Topdown) error {
+			key := PlugTerm(comp.Key, child.Binding)
+			value := PlugTerm(comp.Value, child.Binding)
+			result = append(result, [2]*ast.Term{key, value})
```

Comment:
We should check for conflicts here. For example:

```
s = {["a", 1], ["b", 2], ["a", 3]}
m = {k: v | s[x]; x = [k, v]}
```

Need to update the docs as well to explain this behaviour.